stdlib/distutils/command/config.pyi

```diff
@@ -80,4 +81,4 @@ class config(Command):
         self, header: str, include_dirs: Sequence[str] | None = None, library_dirs: Sequence[str] | None = None, lang: str = "c"
     ) -> bool: ...
 
-def dump_file(filename: str, head: Any | None = None) -> None: ...
+def dump_file(filename: FileDescriptorOrPath, head: Any | None = None) -> None: ...
```

Comment:
File descriptors don't work with the `log.info('%s', filename)` line in the implementation. I suppose we could overload on the value of `head` but that seems like overkill.

Reply:
Doesn't printf-style `'%s' % value` just coerce to string? https://github.com/python/cpython/blob/v3.11.9/Lib/distutils/log.py#L25

And yeah if `FileDescriptorOrPath` doesn't fit all code paths for a method, it's overkill to overload it.

Reply:
Oh right, I've been writing too much C where this would segfault. I guess we can allow file descriptors here if they would work at runtime, but I'm sort of dubious on how helpful it is. There's a benefit to adding fds to the allowed types in our stubs (someone could be intentionally passing in an fd) but also a cost (someone could accidentally pass in some random int and now they won't see a type error), and I'm not convinced the benefits outweigh the costs.

Then again, I don't really use distutils, so I'm happy to defer to your judgment.

Reply:
> I don't really use distutils, so I'm happy to defer to your judgment.

I care most about allowing PathLike wherever possible, even if it's an implementation detail. *especially* for methods used by, or overridden by, setuptools.
`FileDescriptorOrPath` is candy on top.

> but also a cost (someone could accidentally pass in some random int and now they won't see a type error), and I'm not convinced the benefits outweigh the costs.
> but I would personally skip some of the overly precise types that seem to be encoding implementation details.

You make a good point. Eh, I'm not attached to it. I won't keep the filedescriptors params